src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Represents the file used to load and save data.
+ */
+public class Storage {
+
+    private String filepath;
+
+    /**
+     * A constructor to create a Storage object.
+     *
+     * @param filepath The location of the file to load from and save to.
+     */
+    Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Reads the list of tasks from a file.
+     *
+     * @param ls A list to keep track of all tasks.
+     */
+    public void readFile(TaskList ls) {
```

Comment:
method is abit too long and has quite a number of nested statements. you can create another method to convert the text format in storage to tasks.